flytekit/core/map_task.py

```diff
@@ -263,25 +264,36 @@ def _raw_execute(self, **kwargs) -> Any:
             outputs_expected = False
         outputs = []
 
-        mapped_input_value_len = 0
+        mapped_tasks_count = 0
         if self._run_task.interface.inputs.items():
             for k in self._run_task.interface.inputs.keys():
                 v = kwargs[k]
                 if isinstance(v, list) and k not in self.bound_inputs:
-                    mapped_input_value_len = len(v)
+                    mapped_tasks_count = len(v)
                     break
 
-        for i in range(mapped_input_value_len):
+        failed_count = 0
+        min_successes = mapped_tasks_count
+        if self._min_success_ratio:
+            min_successes = math.ceil(min_successes * self._min_success_ratio)
+
+        for i in range(mapped_tasks_count):
             single_instance_inputs = {}
             for k in self.interface.inputs.keys():
                 v = kwargs[k]
                 if isinstance(v, list) and k not in self.bound_inputs:
                     single_instance_inputs[k] = kwargs[k][i]
                 else:
                     single_instance_inputs[k] = kwargs[k]
-            o = exception_scopes.user_entry_point(self._run_task.execute)(**single_instance_inputs)
-            if outputs_expected:
-                outputs.append(o)
+            try:
+                o = exception_scopes.user_entry_point(self._run_task.execute)(**single_instance_inputs)
+                if outputs_expected:
+                    outputs.append(o)
+            except Exception as exc:
+                outputs.append(None)
+                failed_count += 1
+                if mapped_tasks_count - failed_count < min_successes:
+                    raise exc
```

Comment:
Can we add a log here, like  `logger.error("The number of successful tasks is lower than the minimum ratio")`

Reply:
ok. updated it.